src/backend/auxil/range_alloc.rs

```diff
@@ -23,11 +27,13 @@ where
         }
     }
 
-    pub fn allocate_range(&mut self, length: T) -> Option<Range<T>> {
+    pub fn allocate_range(&mut self, length: T) -> Result<Range<T>, RangeAllocationError<T>> {
         assert_ne!(length + length, length);
         let mut best_fit: Option<(usize, Range<T>)> = None;
+        let mut fragmented_free_length = length - length;
```

Comment:
This looks like code that could be made more clear by using traits from the `num` crate.  Specifically [this](https://docs.rs/num/0.2.0/num/trait.Zero.html) would be of use here.

Reply:
`num-traits` is also a good crate if you don't want the other goodies it comes with.